// llama-node/dist/llm/rwkv-cpp.js
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/llm/rwkv-cpp.ts
import {
  InferenceResultType,
  Rwkv
} from "@llama-node/rwkv-cpp";

// src/llm/type.ts
var LLMError = class extends Error {
  constructor({
    message,
    tokens,
    completed,
    type
  }) {
    super(message);
    this.tokens = tokens;
    this.completed = completed;
    this.type = type;
  }
};

// src/llm/rwkv-cpp.ts
var RwkvCpp = class {
  load(config) {
    return __async(this, null, function* () {
      const { modelPath, tokenizerPath, nThreads, enableLogging } = config;
      this.instance = yield Rwkv.load(
        modelPath,
        tokenizerPath,
        nThreads,
        enableLogging
      );
    });
  }
  createCompletion(params, callback, abortSignal) {
    return __async(this, null, function* () {
      let completed = false;
      const tokens = [];
      const errors = [];
      return new Promise(
        (res, rej) => {
          const abort = this.instance.inference(params, (response) => {
            var _a;
            switch (response.type) {
              case InferenceResultType.Data: {
                const data = {
                  token: response.data.token,
                  completed: !!response.data.completed
                };
                tokens.push(data.token);
                if (data.completed) {
                  completed = true;
                }
                callback(data);
                break;
              }
              case InferenceResultType.End: {
                if (errors.length) {
                  rej(
                    new LLMError({
                      message: errors.join("\n"),
                      tokens,
                      completed,
                      type: "Generic" /* Generic */
                    })
                  );
                } else {
                  res({ tokens, completed });
                }
                break;
              }
              case InferenceResultType.Error: {
                errors.push((_a = response.message) != null ? _a : "Unknown Error");
                break;
              }
            }
          });
          const abortSignalHandler = () => {
            abort();
            rej(
              new LLMError({
                message: "Aborted",
                tokens,
                completed,
                type: "Aborted" /* Aborted */
              })
            );
            abortSignal == null ? void 0 : abortSignal.removeEventListener(
              "abort",
              abortSignalHandler
            );
          };
          abortSignal == null ? void 0 : abortSignal.addEventListener("abort", abortSignalHandler);
        }
      );
    });
  }
  // embedding not implemented yet
  /* async getEmbedding(params: RwkvInvocation): Promise<number[]> {
          return new Promise<number[]>((res, rej) => {
              this.instance.getWordEmbedding(params, (response) => {
                  switch (response.type) {
                      case EmbeddingResultType.Data:
                          res(response.data ?? []);
                          break;
                      case EmbeddingResultType.Error:
                          rej(new Error("Unknown Error"));
                          break;
                  }
              });
          });
      }
  
      async getDefaultEmbedding(text: string): Promise<number[]> {
          return this.getEmbedding({
              topP: 0.1,
              temp: 0.1,
              prompt: text,
              maxPredictLength: 2048,
          });
      } */
  tokenize(params) {
    return __async(this, null, function* () {
      return yield this.instance.tokenize(params.content);
    });
  }
};
export {
  RwkvCpp
};
